### PR TITLE
Fix unintentional locale dependency

### PR DESCRIPTION
**Pull request by behdad**
Supersedes https://github.com/harfbuzz/harfbuzz/pull/3357

### CODE REVIEW

**Comment by matthiasclasen** [comment]
Looks fine to me. Didn't know about this lazy loader thing

**Review by matthiasclasen — APPROVED** [review]

File: src/hb-common.cc [hunk 1]
```diff
@@ -1039,6 +1043,56 @@ hb_variation_from_string (const char *str, int len,
   return false;
 }
 
+#if !defined(HARFBUZZ_NO_SETLOCALE) && defined(HAVE_NEWLOCALE) && defined(HAVE_USELOCALE)
```
Comment (drott) [inline src/hb-common.cc:1046]: Should this be `HB_NO_SETLOCALE`?

IIUC, if I set, `HB_NO_SETLOCALE`, it looks like I still need to include a `locale.h` or `xlocale.h` on Linux and Mac respectively in order to find a `locale_t` even though the calls are a no-op? Can that be avoided?

Comment (behdad) [reply src/hb-common.cc:1046]: Fixed. Thanks.

**Comment by drott** [comment]
Thank you!

**Pull request merged**